web-bundle/src/main/java/com/graphhopper/resources/MapboxResponseConverter.java

```diff
@@ -0,0 +1,323 @@
+        if (modifier != null)
+            maneuver.put("modifier", modifier);
+
+        maneuver.put("type", getTurnType(instruction, index));
+        // exit number
+        if (instruction instanceof RoundaboutInstruction)
+            maneuver.put("exit", ((RoundaboutInstruction) instruction).getExitNumber());
+
+        maneuver.put("instruction", instruction.getTurnDescription(trMap.getWithFallBack(locale)));
+
+    }
+
+    /**
+     * Relevant maneuver types are:
+     * depart (firs instruction)
+     * turn (regular turns)
+     * roundabout (enter roundabout, maneuver contains also the exit number)
+     * arrive (last instruction and waypoints)
+     *
+     * You can find all at: https://www.mapbox.com/api-documentation/#maneuver-types
+     */
+    private static String getTurnType(Instruction instruction, int index) {
+        if (index == 0) {
+            return "depart";
+        } else {
+            switch (instruction.getSign()) {
+                case Instruction.FINISH:
+                case Instruction.REACHED_VIA:
+                    return "arrive";
+                case Instruction.USE_ROUNDABOUT:
+                    return "roundabout";
+                default:
+                    return "turn";
+            }
+        }
+    }
+
+    /**
+     * No modifier values for arrive and depart
+     *
+     * Find modifier values here: https://www.mapbox.com/api-documentation/#stepmaneuver-object
+     */
+    private static String getModifier(Instruction instruction) {
+        switch (instruction.getSign()) {
+            case Instruction.CONTINUE_ON_STREET:
+                return "straight";
+            case Instruction.U_TURN_LEFT:
+            case Instruction.U_TURN_RIGHT:
+            case Instruction.U_TURN_UNKNOWN:
+                return "uturn";
+            case Instruction.KEEP_LEFT:
+            case Instruction.TURN_SLIGHT_LEFT:
+                return "slight left";
+            case Instruction.TURN_LEFT:
+                return "left";
+            case Instruction.TURN_SHARP_LEFT:
+                return "sharp left";
+            case Instruction.KEEP_RIGHT:
+            case Instruction.TURN_SLIGHT_RIGHT:
+                return "slight right";
+            case Instruction.TURN_RIGHT:
+                return "right";
+            case Instruction.TURN_SHARP_RIGHT:
+                return "sharp right";
+            case Instruction.USE_ROUNDABOUT:
+                // TODO: We should calculate this (maybe via angle?)
+                return "straight";
+            default:
+                return null;
+        }
+    }
+
+    /**
+     * Puts a location array in GeoJson format into the node
+     */
+    private static ObjectNode putLocation(double lat, double lon, ObjectNode node) {
+        ArrayNode location = node.putArray("location");
+        // GeoJson lon,lat
+        location.add(Helper.round6(lon));
+        location.add(Helper.round6(lat));
```

Comment:
Maybe this is also dependent on the precision asked from outside? (I think I saw something for the simplification)

Reply:
No, this is only for the polyline. The request parameter is `geometries=polyline6` and only changes the precision of the Google Polyline like encoding.